app/scripts/modules/core/src/presentation/icons/Icon.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { memo } from 'react';
+
+import { iconsByName } from './iconsByName';
+
+export type IconNames = keyof typeof iconsByName;
+
+export interface IIconProps {
+  name: IconNames;
+  color: string;
+  customColor?: string;
+  size?: 'small' | 'medium' | 'large';
+  customSize?: string;
+  className?: string;
+}
+
+const DEFAULT_SIZE = 'small';
+
+const pxDimensionsBySize = {
+  small: '24px',
+  medium: '34px',
+  large: '40px',
```

Comment:
@gcomstock In thinking about the API for this component I wanted to encourage more standardized/opinionated practices around icons. A couple questions for you:

- Do you feel confident enough about how icons will be used _going forward_ (don't worry about existing icon usage) to make choices about a standard set of 'sizes' like this? It's not impossible to backtrack/change our mind, but it will be much much harder to do so after we've migrated a bunch of font file icons over
- If the answer to the first question is yes, do you feel these _specific_ names and sizes are appropriate? I went off of what I've seen in both mockups and prototype code

Reply:
Are there _types_ of icons that can inform the color?  This could be restrictive as far as colors go, but also worth thinking about. I see component libraries take this approach

Reply:
@erikmunson The key to making an icon jive with text is to have it hold a similar visual weight. I prefer using the light/"hollow" icons most of the time because the stroke width is similar to a text's stroke width, so your eye doesn't see them as two things smushed together but instead one thing.

Having said that, there are times when an icon also needs to be visually substantial and hold its own, so it's not just going to be light icons, but also bolder ones. So it's not just a matter of size, but also weight. But it's not like we should include multiple weights of every icon. It's circumstantial.

If we were to make some rules, I'd suggest the following:

x-small: 16
small: 20
medium: 24
large: 32
x-large: 40

I think this will cover almost every use case, except for when we want to use an svg icon as an image. Meaning, visually it doesn't read like an icon, but is instead given the prominence and importance of a picture. I think if we get much larger though it's just generally more fun to use more visually rich svgs, like the characters I've used before.

@caseyhebebrand This is a cool idea! I haven't considered it. My goal with the icon color itself is readability, which usually means contrast against its background, vs having the icon color communicate something more. ex: a white icon against a red/green background I feel more quickly communicates bad/good instead of a red/green icon against a white background. We're starting to wade into subjective waters here though.

If we want to get prescriptive with the icon color, we can maybe just have:

dark: #000
neutral: #777
light: #fff

I scanned over my designs and I think it satisfies almost every use case.

Thanks!

Reply:
actually I think using neutral: rgba(0,0,0,0.5) would make it a bit more foolproof. Thanks!

Reply:
I am _super_ into that expanded set of sizes. Thanks for the brainpower.

> I think this will cover almost every use case, except for when we want to use an svg icon as an image.

I agree, and I'm explicitly trying to make SVG images out of scope for the job of this component — IMO being very prescriptive about the API makes it possible to avoid this getting abused for graphics/images (which should really just use the existing pattern we already have in deck today).

Thanks to both of you for the feedback/suggestions around getting even more prescriptive with color — based on the direction of Greg's work overall I am also in favor of providing a dark/neutral/light option. In order to make this component work for the current state of deck in addition to where we want to go, what about providing that more prescriptive API as a first class thing, and demoting `color` to a sort of custom/escape hatch property? For example:

```tsx
<Icon name="artifacts" appearance="light" />
```

```tsx
<Icon name="artifacts" color="alabaster" />
```

We could also make `color` accept both `light`/`neutral`/`dark` _and_ color strings, but I believe I have a slight preference for keeping them separate so that examples, guides, newer code etc. all push folks down a path where they're using those three options, rather than giving them a big bag of strings to pick from.

Reply:
Ok so the new API I've published here looks more like this:

```tsx
// 'neutral' appearance, 'small' size
<Icon name ="artifact" />

// This would be the new-world, preferred way to use icons
<Icon name ="artifact" appearance="light" size="medium" />

// these explicit color and size props would be legacy ways
// of migrating the rest of deck, we could refactor them away
// over time or keep them
<Icon name ="artifact" color="danger" size="21px" />

```